quic/s2n-quic-platform/src/socket/io/tx.rs

```diff
@@ -44,32 +45,36 @@ impl<T: Message> tx::Tx for Tx<T> {
             return Poll::Pending;
         }
 
-        let mut is_any_ready = false;
-        let mut is_all_closed = true;
-
-        for channel in &mut self.channels {
-            match channel.poll_acquire(1, cx) {
-                Poll::Ready(_) => {
-                    is_all_closed = false;
-                    is_any_ready = true;
-                }
-                Poll::Pending => {
-                    is_all_closed &= !channel.is_open();
+        // NOTE: we don't wrap the above check in the contract as we'd technically violate the
+        // contract since we're returning `Pending` without storing a waker
```

Comment:
interesting that there are exceptions to the contract, we'd need some annotation or something to allow for exceptions if we wanted to enforce usage of contract_debug in CI or something

Reply:
Yeah this one is interesting. We don't want to wake the endpoint any time there is `tx` interest. We really only want to poll the inner socket if we have at least 1 item waiting to be sent. So this is a bit of a hack to return `Pending` until we have an item to send.